src/Application/Actions/Person/Update.php

```diff
@@ -0,0 +1,149 @@
+ *         @OA\JsonContent(ref="#/components/schemas/Person"),
+ *     ),
+ *     @OA\Response(
+ *         response=400,
+ *         description="Invalid or missing data",
+ *         @OA\JsonContent(
+ *          format="object",
+ *          example={
+ *              "error": {
+ *                  "description": "The error details",
+ *              }
+ *          },
+ *         ),
+ *     ),
+ *     @OA\Response(
+ *         response=401,
+ *         description="JWT token verification failed",
+ *     ),
+ * ),
+ */
+class Update extends Action
+{
+    public function __construct(
+        LoggerInterface $logger,
+        private readonly PersonRepository $personRepository,
+        private readonly SerializerInterface $serializer,
+        private readonly StripeClient $stripeClient,
+        private readonly ValidatorInterface $validator,
+    ) {
+        parent::__construct($logger);
+    }
+
+    /**
+     * @return Response
+     * @throws HttpBadRequestException
+     * @throws HttpNotFoundException
+     */
+    protected function action(): Response
+    {
+        $person = $this->personRepository->find($this->request->getAttribute('personId'));
+        if (!$person) {
+            throw new HttpNotFoundException($this->request, 'Person not found');
+        }
+
+        try {
+            /** @var Person $person */
+            $person = $this->serializer->deserialize(
+                $body = ((string) $this->request->getBody()),
+                Person::class,
+                'json'
+            );
+        } catch (UnexpectedValueException | TypeError $exception) {
+            // UnexpectedValueException is the Serializer one, not the global one
+            $this->logger->info(sprintf('%s non-serialisable payload was: %s', __CLASS__, $body));
+
+            $message = 'Person Update data deserialise error';
+            $exceptionType = get_class($exception);
+
+            return $this->validationError(
+                "$message: $exceptionType - {$exception->getMessage()}",
+                $message,
+                empty($body), // Suspected bot / junk traffic sometimes sends blank payload.
+            );
+        }
+
+        $violations = $this->validator->validate($person, null, ['complete']);
+
+        if (count($violations) > 0) {
+            $message = 'Validation error: ';
+
+            $violationDetails = [];
+            foreach ($violations as $violation) {
+                $violationDetails[] = $this->summariseConstraintViolation($violation);
+            }
+
+            $message .= implode('; ', $violationDetails);
+
+            return $this->validationError(
+                $message,
+                null,
+                true,
+            );
+        }
+
+        $person = $this->personRepository->persist($person);
+
+        $customerDetails = [
+            'email' => $person->email_address,
+            'name' => sprintf('%s %s', $person->first_name, $person->last_name),
```

Comment:
@NoelLH Similarly, do we check that `first_name` and `last_name` are not null?

Reply:
As above